pkg/proxy/origins/clickhouse/clickhouse.go

```diff
@@ -90,39 +90,34 @@ func (c *Client) Router() http.Handler {
 
 // ParseTimeRangeQuery parses the key parts of a TimeRangeQuery from the inbound HTTP Request
 func (c *Client) ParseTimeRangeQuery(r *http.Request) (*timeseries.TimeRangeQuery, error) {
-
-	trq := &timeseries.TimeRangeQuery{Extent: timeseries.Extent{}}
-	trq.TemplateURL = urls.Clone(r.URL)
-	qi := trq.TemplateURL.Query()
+	qi := r.URL.Query()
+	var rawQuery string
 	if p, ok := qi[upQuery]; ok {
-		trq.Statement = p[0]
+		rawQuery = p[0]
 	} else {
 		return nil, errors.MissingURLParam(upQuery)
 	}
 
-	mp := []string{"step", "timeField"}
-	found := matching.GetNamedMatches(reTimeFieldAndStep, trq.Statement, mp)
-
-	for _, f := range mp {
-		v, ok := found[f]
-		if !ok || v == "" {
-			return nil, errors.ErrNotTimeRangeQuery
-		}
-		switch f {
-		case "timeField":
-			trq.TimestampFieldName = v
-		case "step":
-			trq.Step, _ = tt.ParseDuration(v + "s")
-		}
+	var bf time.Duration
+	res := request.GetResources(r)
+	if res == nil {
+		bf = 60 * time.Second
+	} else {
+		bf = res.OriginConfig.BackfillTolerance
 	}
 
-	var err error
-	trq.Statement, trq.Extent, _, err = getQueryParts(trq.Statement, trq.TimestampFieldName)
-	if err != nil {
+	// Force gzip compression since Brotli is broken on CH 20.3
+	// See https://github.com/ClickHouse/ClickHouse/issues/9969
+	// Clients that don't understand gzip are going to break, but oh well
+	r.Header.Set("Accept-Encoding", "gzip")
```

Comment:
We have a pretty cool way to set headers like this - check out routes.go file in this package, and where we have the main/only path defined in the DefaultPathConfigs(). I think this will work:

```go
// in clickhouse/routes.go

import "github.com/tricksterproxy/trickster/pkg/proxy/headers" // <-- add to imports

paths := map[string]*po.Options{
	"/": {
		Path:           "/",
		HandlerName:    "query",
		Methods:        []string{http.MethodGet, http.MethodPost},
		MatchType:      matching.PathMatchTypePrefix,
		MatchTypeName:  "prefix",
		CacheKeyParams: []string{"query", "database"},
		RequestHeaders: map[string]string{headers.NameAcceptEncoding: "gzip"}, // <-- the new line
	},
}
```

That way we do it in a known place, and it's easily/directly overridden by config.

Reply:
so I tried this, and Grafana blew up?  I wonder if we were sending two Accept-Encoding headers and ClickHouse got confused and sent back Brotli but said it was sending back gzip (that's kind of what it looked like, Trickster logs got horribly ugly too).  I'm happy to revisit this since using the route definition is much cleaner, but the whole thing is really a hack around broken ClickHouse Brotli and I'm hoping they'll fix the bug and I can remove it completely, so I don't have a lot of motivation to figure out the details.